pallets/identity/src/lib.rs

```diff
@@ -167,7 +153,45 @@ decl_module! {
                 WithdrawReason::Fee.into(),
                 ExistenceRequirement::KeepAlive,
             )?;
-            Self::_register_did(sender, signing_items)
+
+            let _new_id = Self::_register_did(sender, signing_items)?;
+            Ok(())
+        }
+
+        /// Register `target_account` with a new Identity.
+        ///
+        /// # Failure
+        /// - `origin` has to be a trusted KYC provider.
+        /// - `target_account` (master key of the new Identity) can be linked to just one and only
+        /// one identity.
+        /// - External signing keys can be linked to just one identity.
+        pub fn cdd_register_did(
+            origin,
+            target_account: T::AccountId,
+            kyc_claim_expiry: T::Moment,
+            kyc_claim: ClaimValue,
+            signing_items: Vec<SigningItem>) -> DispatchResult {
+
+            // Sender has to be part of KYCProviders
+            let cdd_sender = ensure_signed(origin)?;
+            let cdd_key = AccountKey::try_from(cdd_sender.encode())?;
+            let cdd_id = Context::current_identity_or::<Self>(&cdd_key)?;
+
+            let kyc_providers = T::KycServiceProviders::get_members();
+            ensure!( kyc_providers.into_iter().any( |kyc_id| kyc_id == cdd_id),
+                Error::<T>::UnAuthorizedKYCProvider);
+
+            // Subtract fee from target: It will need an authoriztion from `target_account`.
+            let _imbalance = <T::Balances>::withdraw(
+                &cdd_sender,
+                Self::did_creation_fee(),
+                WithdrawReason::Fee.into(),
+                ExistenceRequirement::KeepAlive,
+            )?;
```

Comment:
Since we are not handling the imbalance here, this will leave a hold in the total supply. We are reducing someone's balance but not increasing anyone's else balance or decreasing total supply. This will mean that the sum of all balances will become less than the total supply. Perhaps we should add a TODO here and tackle this in a separate task.

Reply:
Agree on the TODO approach. It's possible we'll remove this fee, and we need to handle all module fees consistently.